### ISSUE TITLE
Re-created service account does not trigger replacement of PubSub IAM bindings

### ISSUE THREAD

**Issue by brabster** [terraform-providers/terraform-provider-google#6532]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
Terraform v0.12.25

### Affected Resource(s)

* google_service_account (data source)

### Terraform Configuration Files

```tf
data "google_service_account" "publisher_service_account" {
  account_id =var.account_id
  project = var.project
}

resource "google_pubsub_topic_iam_member" "pubsub_publisher" {
  topic = var.topic_name
  role = "roles/pubsub.publisher"
  member = "serviceAccount:${data.google_service_account.publisher_service_account.email}"
}
```

### Debug Output

Cannot share debug output from this project. Would need to recreate a standalone example if needed.

### Panic Output

Did not panic

### Expected Behavior

If the service account has been re-created with the same email address, it gets a new UID. Existing bindings no longer work (as they internally refer to the original UID). The datasource for the service account should be aware of the change in UID, so should produce a plan to replace the bindings.

### Actual Behavior

Plan was a no-op.

### Steps to Reproduce

1. create a service account and a topic
2. Terraform a publish binding between the service account and the topic.
3. Delete and re-create the service account with the same email address
4. Terraform plan

### Important Factoids

In this specific case the SA was in a different project, but it probably doesn't need to be.

### References

* https://cloud.google.com/iam/docs/understanding-service-accounts#deleting_and_recreating_service_accounts

**Comment by edwardmedia** [comment]
@brabster how did you delete and recreate the service account? If you did via other tools, Terraform does not have knowledge about this change. Do you see your expected behavior somewhere?

**Comment by brabster** [comment]
Hi @edwardmedia, thanks forthe question! The account was deleted and recreated by Terraform, but a totally independent Terraform state. I raised this issue because I could see the updated UID when imported the service account, so it seemed like the provider could detect the change and handle it as a replacement if it stored the original UID of a service account in the TF state.

As I've learned since, the utility is potentially limited by the need for permissions to get the service account details, so I'm not sure the feature would be useful in general.

If I come across the issue again, I'll just re-create the service account with a different email address so keep things simple!
